bundle/config/interpolation/interpolation.go

```diff
@@ -17,7 +17,8 @@ import (
 
 const Delimiter = "."
 
-var re = regexp.MustCompile(`\$\{(\w+(\.\w+)*)\}`)
+// must start with alphabet, support hyphens and underscores in middle but must end with character
+var re = regexp.MustCompile(`\$\{([a-zA-Z]+([-_]*[a-zA-Z0-9]+)*(\.[a-zA-Z]+([-_]*[a-zA-Z0-9]+)*)*)\}`)
```

Comment:
Modified `[-_]?` instead of `[-_]*`, bit more restrictive not allowing consecutive `_` or `-`.

```suggestion
var re = regexp.MustCompile(`\$\{([a-zA-Z]+([-_]?[a-zA-Z0-9]+)*(\.[a-zA-Z]+([-_]?[a-zA-Z0-9]+)*)*)\}`)
```

Reply:
makes sense, the names should be human readable. i will add this to the unit test so there is some documentation that this is a non interpolation case.